src/components/AssetManagementPanel/AssetManagementForm/FormVariants/NominateBeneficiaryHolder/NominateBeneficiaryHolder.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useContext, useEffect, useState } from "react";
+import { FormState } from "../../../../../constants/FormState";
+import { ButtonSolidOrangeWhite, ButtonSolidWhiteGrey } from "../../../../UI/Button";
+import { LoaderSpinner } from "../../../../UI/Loader";
+import { AssetInformationPanel } from "../../../AssetInformationPanel";
+import { AssetManagementActions } from "../../../AssetManagementActions";
+import { AssetManagementTitle } from "../../AssetManagementTitle";
+import { OverlayContext } from "./../../../../../common/contexts/OverlayContext";
+import {
+  MessageTitle,
+  showDocumentTransferMessage,
+} from "./../../../../../components/UI/Overlay/OverlayContent/DocumentTransferMessage";
+import { EditableAssetTitle } from "./../EditableAssetTitle";
+
+interface NominateBeneficiaryHolderFormProps {
+  formAction: AssetManagementActions;
+  tokenId: string;
+  tokenRegistryAddress: string;
+  beneficiary?: string;
+  holder?: string;
+  handleTransfer: (newBeneficiary: string, newHolder: string) => void;
+  nominationState: string;
+  onBack: () => void;
+}
+
+export const NominateBeneficiaryHolderForm = ({
+  formAction,
+  tokenId,
+  tokenRegistryAddress,
+  beneficiary,
+  holder,
+  handleTransfer,
+  nominationState,
+  onBack,
+}: NominateBeneficiaryHolderFormProps) => {
+  const [newBeneficiary, setNewBeneficiary] = useState("");
+  const [newHolder, setNewHolder] = useState("");
+  const isPendingConfirmation = nominationState === FormState.PENDING_CONFIRMATION;
+  const isConfirmed = nominationState === FormState.CONFIRMED;
+  const isEditable = nominationState !== FormState.PENDING_CONFIRMATION && nominationState !== FormState.CONFIRMED;
+  const { showOverlay } = useContext(OverlayContext);
+
+  useEffect(() => {
+    if (isConfirmed) {
+      showOverlay(
+        showDocumentTransferMessage(MessageTitle.NOMINATE_BENEFICIARY_HOLDER_SUCCESS, {
+          isSuccess: true,
+        })
+      );
+    }
+  }, [isConfirmed, newBeneficiary, showOverlay]);
+
+  const isValidEndorse = () => {
+    if (!newBeneficiary || !newHolder) return false;
```

Comment:
Might also want to consider a function that checks if its a valid smart contract address. 
I think the `web3-utils` package does that with `isAddress` (https://www.npmjs.com/package/web3-utils).

Reply:
While at it, might want to add that condition for other forms as well lol

Reply:
yup okay